### PR TITLE
Switch to txrequests to marry async requests with twisted event loop

### PR DESCRIPTION
**Pull request by lyoshenka**

### CODE REVIEW

**Comment by lyoshenka** [comment]
@jackrobison this was surprisingly painless

**Comment by lyoshenka** [comment]
Aaaaand clearly I messed something up. Run `lbrynet-daemon --fake-arg`. Process hangs and needs to be `kill -9`ed

**Comment by lyoshenka** [comment]
Does it have something to do with the order in which twisted and logging are initialized?

**Comment by lyoshenka** [comment]
Of course it does.

File: lbrynet/core/log_support.py [hunk 1]
```diff
@@ -40,6 +40,10 @@ def bg_cb(sess, resp):
 
 class HTTPSHandler(logging.Handler):
     def __init__(self, url, fqdn=False, localname=None, facility=None):
+        global session
+        if session is None:
+            session = Session()
+
```
Comment (jackrobison) [inline lbrynet/core/log_support.py]: Why global, and why in __init__?

File: lbrynet/analytics/api.py [hunk 2]
```diff
@@ -2,33 +2,29 @@
...
+        elif isinstance(response_or_failure, Failure):
+            log.warning('Failed to send an analytics event. Error: {}'.format(str(Failure)))
 
     @functools.wraps(fn)
     def wrapper(*args, **kwargs):
-        future = fn(*args, **kwargs)
-        future.add_done_callback(_log)
-        return future
+        d = fn(*args, **kwargs)
+        d.addBoth(_log)
```
Comment (jackrobison) [inline lbrynet/analytics/api.py]: Why addBoth?

File: lbrynet/analytics/api.py [hunk 3]
```diff
@@ -3,32 +3,24 @@
...
 log = logging.getLogger(__name__)
 
 
 def log_response(fn):
-    def _log(future):
-        if future.cancelled():
-            log.warning('Request was unexpectedly cancelled')
-        elif future.exception():
-            exc, traceback = future.exception_info()
-            log.warning('Failed to send an analytics event', exc_info=(type(exc), exc, traceback))
```
Comment (jackrobison) [inline lbrynet/analytics/api.py]: I think exc_info is necessary to get the real traceback and not a useless trace from twisted internals

**Comment by lyoshenka** [comment]
merged

**Pull request closed without merging**